frontend/src/hooks/useAsyncDebounceEffect.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef } from 'react';
+
+const useAsyncDebounceEffect = (
```

Comment:
Takes useEffect parameters, plus an optional delay parameter
- Staggers the executions of the callback
- Won't execute if callback is in progress
- Perfect for not bashing the API when we have a bunch of typing coming in